src/Core/src/Handlers/Image/ImageHandler.Windows.cs

```diff
@@ -55,6 +57,44 @@ protected override void RemoveContainer()
 			UpdateValue(nameof(IView.Width));
 		}
 
+		/// <inheritdoc/>
+		public override Graphics.Size GetDesiredSize(double widthConstraint, double heightConstraint)
+		{
+			// AspectFit + non-Fill alignment: cap Max* to intrinsic bitmap so alignment (Center/Start/End) works.
+			// Only set once when Max* still Infinity and we have decoded size; else mirror VirtualView Maximum*.
+			// Without this the Image measures to available space and alignment appears ignored.
+			if (VirtualView.Aspect == Aspect.AspectFit
+				&& VirtualView.HorizontalLayoutAlignment != Primitives.LayoutAlignment.Fill
+				&& VirtualView.VerticalLayoutAlignment != Primitives.LayoutAlignment.Fill)
+			{
+				// First (and only) chance to lock to intrinsic size.
+				if (PlatformView.MaxWidth == double.PositiveInfinity
+					&& PlatformView.MaxHeight == double.PositiveInfinity)
+				{
+					// Clamp to decoded pixel size if available.
+					var imageSize = GetImageSize();
+
+					if (imageSize.Width != 0 && imageSize.Height != 0)
+					{
+						PlatformView.MaxWidth = imageSize.Width;
```

Comment:
Chatting with @MartyIX they made some good points about if we want to cause side effects like this when calling GetDesiredSize

Would it be better to instead just call 

var possibleSize =  base.GetDesiredSize(widthConstraint, heightConstraint);

And then compare that possibleSize against the imageSize to see if we want to return a different possibleSize? 

Instead of relying on the Max values of the PlatformView? 

@MartyIX made some good points about GetDesiredSize causing side effects on the platformview might be not desireable.

Reply:
yea, I guess the other approach here would be to create our own MauiImage and override the arrange/measure passes there

But I'd rather not if we don't have to :-)